### PR TITLE
DIRECTOR: LINGO: Implement property array functions

### PR DESCRIPTION
**Pull request by npjg**
This pull request implements Lingo property array functions, and updates other array functions to act appropriately on property arrays. Updates to the tests are forthcoming.

### CODE REVIEW

File: engines/director/lingo/lingo.cpp [hunk 1]
```diff
@@ -578,9 +578,9 @@ const char *Datum::type2str(bool isk) {
 	}
 }
 
-int Datum::compareTo(Datum d) {
+int Datum::compareTo(Datum d, bool ignoreCase) {
 	if (type == STRING && d.type == STRING) {
-		return u.s->compareTo(*d.u.s);
+		return ignoreCase ? u.s->compareToIgnoreCase(*d.u.s) : u.s->compareTo(*d.u.s);
```
Comment (djsrv) [inline engines/director/lingo/lingo.cpp:612]: This should use `toLowercaseMac` as in #2204

Comment (npjg) [reply engines/director/lingo/lingo.cpp]: Fixed, thanks.

**Comment by djsrv** [comment]
This appears to have broken array comparison.

Before when I ran:
```
set a = [0,0]
set b = [1,1]
put a = b
```

This was the result:
```
[ 22]: Stack before: <[0, 0]> <[1, 1]> 
Vars before
  Local vars: a, b, 
  Global vars: mDispose, mPut, mPerform, mName, mInstanceRespondsTo, mDescribe, mNew, mMessageList, mAtFrame, mGet, mRespondsTo, 
[ 22]: c_eq
[ 22]: Stack after: <0> 
Vars after
  Local vars: a, b, 
  Global vars: mDispose, mPut, mPerform, mName, mInstanceRespondsTo, mDescribe, mNew, mMessageList, mAtFrame, mGet, mRespondsTo,
```

Now:
```
[ 22]: Stack before: <[0, 0]> <[1, 1]> 
Vars before
  Local vars: a, b, 
  Global vars: mDispose, mPut, mPerform, mName, mInstanceRespondsTo, mDescribe, mNew, mMessageList, mAtFrame, mGet, mRespondsTo, 
[ 22]: c_eq
Segmentation fault (core dumped)
```

Property array comparison segfaults too.

**Comment by sev-** [comment]
Cool, we addressed all issue offline. Merging.

**Pull request merged**